import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import javax.naming.Context;
import javax.naming.InitialContext;
import javax.naming.NamingException;
import javax.sql.DataSource;

public class UsuarioDao {
    
    private DataSource dataSource;
    
    public UsuarioDao() {
        try {
            Context initContext = new InitialContext();
            Context envContext = (Context) initContext.lookup("java:/comp/env");
            dataSource = (DataSource) envContext.lookup("jdbc/minha-base-dados");
        } catch (NamingException e) {
            throw new RuntimeException("Não foi possível obter o DataSource", e);
        }
    }
    
    public void adicionar(Usuario usuario) {
        String sql = "INSERT INTO usuario (Nome, Email, DataDeNascimento) VALUES (?, ?, ?)";
        
        try (Connection usu = dataSource.getConnection();
                PreparedStatement stmt = usu.prepareStatement(sql)) {
            
            stmt.setString(1, contato.getNome());
            stmt.setString(2, contato.getEmail());
            stmt.setString(3, contato.getDataDeNascimento());
            
            stmt.executeUpdate();
            
        } catch (SQLException e) {
            throw new RuntimeException("Erro ao inserir contato no banco de dados", e);
        }
    }
}
